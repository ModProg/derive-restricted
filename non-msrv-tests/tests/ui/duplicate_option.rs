#[derive(derive_where::DeriveWhere)]
#[derive_where(Default; T)]
enum Test1<T> {
    #[derive_where(default, default)]
    A(T),
}

#[derive(derive_where::DeriveWhere)]
#[derive_where(Default; T)]
enum Test2<T> {
    #[derive_where(default)]
    #[derive_where(default)]
    A(T),
}

fn main() {}
