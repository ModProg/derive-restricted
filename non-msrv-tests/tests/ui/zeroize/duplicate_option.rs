extern crate zeroize_ as zeroize;

#[derive(derive_where::DeriveWhere)]
#[derive_where(Zeroize(drop, drop); T)]
struct Test1<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Zeroize(crate = "zeroize_", crate = "zeroize_"); T)]
struct Test2<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Zeroize(drop, drop, crate = "zeroize_"); T)]
struct Test3<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Zeroize(drop, crate = "zeroize_", crate = "zeroize_"); T)]
struct Test4<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Zeroize(crate = "zeroize_", crate = "zeroize_", drop); T)]
struct Test5<T>(T);


#[derive(derive_where::DeriveWhere)]
#[derive_where(Zeroize(crate = "zeroize_", drop, drop); T)]
struct Test6<T>(T);

fn main() {}
