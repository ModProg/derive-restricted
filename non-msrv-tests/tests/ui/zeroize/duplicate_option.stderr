error: duplicate `drop` option
 --> tests/ui/zeroize/duplicate_option.rs:4:30
  |
4 | #[derive_where(Zeroize(drop, drop); T)]
  |                              ^^^^

error: duplicate `crate` option
 --> tests/ui/zeroize/duplicate_option.rs:8:44
  |
8 | #[derive_where(Zeroize(crate = "zeroize_", crate = "zeroize_"); T)]
  |                                            ^^^^^^^^^^^^^^^^^^

error: duplicate `drop` option
  --> tests/ui/zeroize/duplicate_option.rs:12:30
   |
12 | #[derive_where(Zeroize(drop, drop, crate = "zeroize_"); T)]
   |                              ^^^^

error: duplicate `crate` option
  --> tests/ui/zeroize/duplicate_option.rs:16:50
   |
16 | #[derive_where(Zeroize(drop, crate = "zeroize_", crate = "zeroize_"); T)]
   |                                                  ^^^^^^^^^^^^^^^^^^

error: duplicate `crate` option
  --> tests/ui/zeroize/duplicate_option.rs:20:44
   |
20 | #[derive_where(Zeroize(crate = "zeroize_", crate = "zeroize_", drop); T)]
   |                                            ^^^^^^^^^^^^^^^^^^

error: duplicate `drop` option
  --> tests/ui/zeroize/duplicate_option.rs:25:50
   |
25 | #[derive_where(Zeroize(crate = "zeroize_", drop, drop); T)]
   |                                                  ^^^^
